patroni/__init__.py

```diff
@@ -145,3 +141,42 @@ def main():
         else:
             patroni.ha.while_not_sync_standby(lambda: patroni.postgresql.stop(checkpoint=False))
             patroni.dcs.delete_leader()
+
+
+def main():
+    if os.getpid() != 1:
+        return patroni_main()
+
+    pid = 0
+
+    # Looks like we are in a docker, so we will act like init
+    def sigchld_handler(signo, stack_frame):
+        try:
+            while True:
+                ret = os.waitpid(-1, os.WNOHANG)
+                if ret == (0, 0):
+                    break
+                elif ret[0] != pid:
+                    logging.info('Reaped unknow pid=%s, exit status=%s', *ret)
+        except OSError:
+            pass
+
+    def passtochild(signo, stack_frame):
+        if pid:
+            os.kill(pid, signo)
+
+        if signo in (signal.SIGTERM, signal.SIGINT, signal.SIGQUIT):
+            sys.exit(0)
```

Comment:
It will pass everything listed on the lines 172-177. And @alexeyklyukin  is right about `sys.exit`, we should not call it, but wait until the child process will exit on the line 182. Fix is coming.